odfe-cli/it/ad_test.go

```diff
@@ -0,0 +1,347 @@
+// +build integration
+
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package it
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"es-cli/odfe-cli/client"
+	adctrl "es-cli/odfe-cli/controller/ad"
+	"es-cli/odfe-cli/controller/es"
+	"es-cli/odfe-cli/entity"
+	adentity "es-cli/odfe-cli/entity/ad"
+	adgateway "es-cli/odfe-cli/gateway/ad"
+	esg "es-cli/odfe-cli/gateway/es"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+//ADTestSuite suite specific to AD plugin
+type ADTestSuite struct {
+	ODFECLISuite
+	DetectorRequest adentity.CreateDetectorRequest
+	Detector        adentity.CreateDetector
+	DetectorId      string
+	ADGateway       adgateway.Gateway
+	ESController    es.Controller
+}
+
+func getRawFeatureAggregation() []byte {
+	return []byte(`{
+        			"sum_value": {
+          				"sum": {
+            				"field": "total_quantity"
+						}
+        			}
+      			}`)
+}
+
+//SetupSuite runs once for every test suite
+func (a *ADTestSuite) SetupSuite() {
```

Comment:
Should this be a method encapsulated in an interface for ODFECLISuite? I assume that all of the Controllers will need to do this in some form.

Reply:
ODFECLISuite will contains details like connection and helper methods like create index, but ADTestSuite will have additional methods and parameter to support their test methods. SetupSuite is framework method which will be called by go test . So, every individual plugin should have their own setup suite which runs once every test suite. What do you think?

Reply:
Got it that makes sense thanks.